oidc-service-impl/src/test/java/io/mosip/esignet/services/OpenIdConnectServiceTest.java

```diff
@@ -69,9 +66,44 @@ public void getUserInfo_withValidTransaction_thenPass() {
         oidcTransaction.setClientId("client-id");
         oidcTransaction.setPartnerSpecificUserToken("p-s-u-t");
         oidcTransaction.setEncryptedKyc("encrypted-kyc");
+        oidcTransaction.setDpopBoundAccessToken(false);
+        Mockito.when(cacheUtilService.getUserInfoTransaction(Mockito.anyString())).thenReturn(oidcTransaction);
+        String kyc = openIdConnectService.getUserInfo("Bearer access-token", null);
+        Assert.assertNotNull(kyc);
+        Assert.assertEquals("encrypted-kyc", kyc);
+    }
+
+    @Test
+    public void getUserInfo_withValidTransactionAndDpopNonce_thenPass() {
+        OIDCTransaction oidcTransaction = new OIDCTransaction();
+        oidcTransaction.setClientId("client-id");
+        oidcTransaction.setPartnerSpecificUserToken("p-s-u-t");
+        oidcTransaction.setEncryptedKyc("encrypted-kyc");
+        oidcTransaction.setDpopBoundAccessToken(true);
+        String dpopHeader = "dpop header";
         Mockito.when(cacheUtilService.getUserInfoTransaction(Mockito.anyString())).thenReturn(oidcTransaction);
-        String kyc = openIdConnectService.getUserInfo("Bearer access-token");
+        Mockito.when(tokenService.isValidDpopServerNonce(dpopHeader, oidcTransaction)).thenReturn(true);
+        String kyc = openIdConnectService.getUserInfo("DPoP access-token", dpopHeader);
         Assert.assertNotNull(kyc);
         Assert.assertEquals("encrypted-kyc", kyc);
     }
+
+    @Test
+    public void getUserInfo_withDpopWithoutNonce_thenFail() throws Exception {
+        OIDCTransaction oidcTransaction = new OIDCTransaction();
+        oidcTransaction.setClientId("client-id");
+        oidcTransaction.setPartnerSpecificUserToken("p-s-u-t");
+        oidcTransaction.setEncryptedKyc("encrypted-kyc");
+        oidcTransaction.setDpopBoundAccessToken(true);
+        Mockito.when(cacheUtilService.getUserInfoTransaction(Mockito.anyString())).thenReturn(oidcTransaction);
+        Mockito.when(tokenService.isValidDpopServerNonce(Mockito.anyString(), Mockito.any())).thenReturn(false);
+        String nonce = "valid-nonce";
+        Mockito.doThrow(new DpopNonceMissingException(nonce)).when(tokenService).generateAndStoreNewNonce(Mockito.anyString(), Mockito.anyString());
+        try {
+            openIdConnectService.getUserInfo("DPoP access-token", "dpop-header");
```

Comment:
After this statement we should have Assert.fail(), otherwise unexpected test results will go unnoticed